### DISCUSSION TITLE
Don't check for updates during startup?

### DISCUSSION THREAD

**Discussion by enoch85** [d2phap/ImageGlass#724] [General] [answered]
### System information:
 - Windows OS version: Windows Server 2012 and 2016
 - ImageGlass version: 7.0.7.26

### Questions
Thanks for a great app! I found it when I was looking for a way to replace Paint when showing picture files in Windows Server 2016 - because in 2016 they removed the regular photo viewer.

We installed it on different session hosts (WS 2016 and 2012) were there are quite many non-technical users. To be able to install the lastest version you need Net Framework 4.7, which is not default in Windows and requires restarts and such. Not a good idea in a production environment with several users online. So, I installed an older version to come around it.

Now, my question is, would it be possible (through a GPO, or during installation or similar) to avoid making ImageGlass check for updates on each startup? No issue for me, but for the end users. Especially when they are many, and very non-technical. Even if we would tell them that it's just changing an option in the settings, we would get a lot of questions on *how* to actually do that. It's also not that nice to install a program, then ask the end-user to do changes to it imho.

Thank you in advance.

**Comment by d2phap** [comment]
Hi @enoch85 

Thanks for your question.

> We installed it on different session hosts (WS 2016 and 2012) were there are quite many non-technical users. To be able to install the latest version you need Net Framework 4.7

We have been technically using .NET Framework 4.7.1 since version 4.0, but in the latest version, I used the Advanced Installer to pack the MSI file, and add requirement for checking .NET Framework 4.7.1. I will remove this check in the next releases.

> Now, my question is, would it be possible (through a GPO, or during installation or similar) to avoid making ImageGlass check for updates on each startup?

Unfortunately, this feature only supports in v7.5 by introducing new configs files, you can read more at: https://imageglass.org/docs/app-configs

There is known issue of IG 7.5 about performance due the the latest version of ImageMagick.NET library, reported at https://github.com/d2phap/ImageGlass/issues/720. I plan to publish a new version when they fix this bug.

**Comment by enoch85** [comment]
> I will remove this check in the next releases.

So what you're saying is that it should be possible to download and install the latest version without having .NET 4.7.1 installed on the server/computer?

**Comment by enoch85** [comment]
> https://imageglass.org/docs/app-configs

Regarding that, what's the specific setting for not checking updates upon start of the app?

I couldn't find a list with available options.

Thanks!

**Comment by d2phap** [comment, marked as answer]
Hi @enoch85 

> So what you're saying is that it should be possible to download and install the latest version without having .NET 4.7.1 installed on the server/computer?

The MSI installer checks for .NET 4.7.1, but the app itself does not.
If you use MSI installer, you may not.


> Regarding that, what's the specific setting for not checking updates upon start of the app?

You can set the `AutoUpdate` to `0`.

For all settings, refer to:
https://github.com/d2phap/ImageGlass/blob/master/Source/Components/ImageGlass.Settings/Configs.cs#L342-L345

**Comment by enoch85** [comment]
Thanks a ton!
